packages/apps/job-launcher/server/src/modules/webhook/webhook.dto.ts

```diff
@@ -1,8 +1,37 @@
 import { ChainId } from '@human-protocol/sdk';
-import { ApiProperty } from '@nestjs/swagger';
-import { IsEnum, IsString, IsBoolean, IsOptional } from 'class-validator';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { IsValidEthereumAddress } from '../../common/validators/ethers';
+import {
+  IsEnum,
+  IsObject,
+  IsString,
+  IsBoolean,
+  IsOptional,
+} from 'class-validator';
 import { EventType, OracleType } from '../../common/enums/webhook';
 
+export class InvalidManifest {
+  @ApiProperty({ name: 'invalid_manifest' })
+  @IsString()
+  invalidManifest: string;
+
+  @ApiProperty()
+  @IsString()
+  reason?: string;
+}
+
+export class ManifestCannotBeDownloaded {
+  @ApiProperty({ name: 'manifest_cannot_be_downloaded' })
+  @IsString()
+  manifestCannotBeDownloaded: string;
+
+  @ApiProperty()
+  @IsString()
+  reason?: string;
+}
+
+export type EventData = InvalidManifest | ManifestCannotBeDownloaded;
```

Comment:
This is not needed. You can use a simple object for both types of evets like:
```suggestion
export class EventData {
  @ApiProperty()
  @IsString()
  reason?: string;
}
```